api/src/main/java/me/lokka30/treasury/api/permission/Context.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/ArcanePlugins/Treasury>.
+ */
+
+package me.lokka30.treasury.api.permission;
+
+import me.lokka30.treasury.api.common.misc.TriState;
+import org.jetbrains.annotations.NotNull;
+
+public final class Context {
+
+    private static final Context GLOBAL = new Context("global", TriState.FALSE);
+
+    @NotNull
+    public static Context global() {
+        return GLOBAL;
+    }
+
+    @NotNull
+    public static Context of(@NotNull String context) {
+        return new Context(context, TriState.UNSPECIFIED);
+    }
+
+    @NotNull
+    public static Context of(@NotNull String context, @NotNull TriState world) {
+        return new Context(context, world);
+    }
+
+    private final String context;
+    private final TriState world;
+
+    private Context(@NotNull String context, @NotNull TriState world) {
+        this.context = context;
+        this.world = world;
+    }
+
+    @NotNull
+    public String context() {
+        return this.context;
+    }
+
+    @NotNull
+    public TriState isWorldContext() {
```

Comment:
Why is this a TriState? What does `UNKNOWN` represent?

What if we want a proxy server context? The current impl only supports world (and global) contexts.

Reply:
`UNKNOWN` represents the global context.

> What if we want a proxy server context? The current impl only supports world (and global) contexts.

world and server are kinda the same thing. it'd be world if it's about a server, and server if it's about a proxy. I guess we need another name for `world`, something between `world` and `server`